tests/security/static/queries/cpp/openenclave/Dereferences.qll

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+/**
+ * Provides classes and predicates for identifying expressions that are dereferences of a pointer.
+ *
+ * This library detects dereferences that occur locally - for example, in a dereferencing
+ * expression, or as an argument to a library call such as memcpy - as well as dereferences that
+ * occur by passing pointer arguments to calls that ultimately dereference the argument.
+ */
+
+import cpp
+import semmle.code.cpp.controlflow.Dereferenced
+
+/**
+ * A pointer expression `e` that is dereferenced directly by `op`.
+ */
+predicate localDereference(Expr op, Expr e) {
+  e.getActualType().getUnspecifiedType() instanceof PointerType and
+  not op = any(SizeofExprOperator sizeof).getAChild*() and
```

Comment:
Does this ignore `sizeof(foo(*ptr))`?

Reply:
In this case ptr is not a direct child of sizeof node so it will not be ignored.
ptr will be considered ad dereference in call (foo) expression node.